test/system-script.test.js

```diff
@@ -1,133 +1,123 @@
 describe('system-amd-script', function() {
-  let validManifestUrl = 'http://localhost:' + window.location.port + '/base/test/manifests/simple.json';
-
-  let system;
+  let SystemJS;
 
   beforeEach(function() {
-    system = new System.constructor();
-    window.system = system;
-    window.SystemJS = system;
+    SystemJS = new System.constructor();
+    window.SystemJS = SystemJS;
     window.define = function () {
       window.canopyDefine.apply(window, arguments);
     }
+    const scripts = document.querySelectorAll('script[data-system-amd-name]')
+    for (let i=0; i<scripts.length; i++) {
+      scripts[i].remove() }
     Object.keys(window.__systemAmdScript.scriptNameMap).forEach(key => {
       delete window.__systemAmdScript.scriptNameMap[key];
     });
   });
 
-  it('should load modules via script tags with meta config', function(done) {
-    system.config({
+  it('should load modules via script tags with meta config', function() {
+    SystemJS.config({
       meta: {
         "navbar": { loader: '/base/test/fixtures/plugin.js' }
       }
     });
 
-    system
-    .import('navbar')
-    .then(function(m) {
-      expect(m()).toBe(1);
-      done();
-    })
-    .catch(fail)
+    return SystemJS
```

Comment:
I ended up refactoring the tests to try to understand them better while I was debugging them. Jasmine and karma-jasmine now support returning promises from tests instead of manually calling done and fail.

Reply:
you are now a true jasmine ninja :)